### PR TITLE
Created the tables and migrations on ROR

### LINKED ISSUE #11: Create the database tables
# REQUIREMENTS
Start creating the database tables and relations in ruby on rails with pg already set up 
Following the diagram
- Employees
- Branch
- Assitence
- Street cp
# DESIGN
Untitled Workspace

### PR DESCRIPTION
**Pull request by juperameza**
# Description
 Examples:
 - What?: I've added the tables and models using ruby on rails syntax  and created the migrations in order to generate a database on postgres
 - Why?: To be able to start creating the cruds with all the relations and tables needed

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ source "https://rubygems.org"
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby "3.0.3"
-gem "pg"
```
Comment (YilKanda) [inline Gemfile:5]: Why did you remove the PostgreSQL gem? You will occupy it when you deploy to Heroku.

Comment (juperameza) [reply Gemfile:5]: Before this we implemented the gem 2 times we needed to delete one in order to avoid conflicts

**Review by JuanequeX — DISMISSED** [review]
LGTM

File: db/migrate/20220310235942_create_admins.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      t.string :name
+      t.string :email
+      t.string :position
+      t.integer :private_number
+      t.string :password
+
+      t.timestamps
+    end
+  end
+end
```
Comment (vyk2rr) [inline db/migrate/20220310235942_create_admins.rb]: I guess this table `:admins` and `:employees` are going to store somehow unique unrepeatable users, right? I mean you can not have two users with the same email but different name, so if that is true, I think as a good practice you have to add indexes to your tables in order to keep that consistency in the data

Comment (YilKanda) [reply db/migrate/20220310235942_create_admins.rb]: I agree with @vyk2rr , since if you implement a session management gem such as [Devise](https://github.com/heartcombo/devise) or [Clearance](https://github.com/thoughtbot/clearance), the gem will automatically generate a user table with the email field, which would generate inconsistency with the email data that you have already generated in your tables.

File: db/migrate/20220310235942_create_admins.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class CreateAdmins < ActiveRecord::Migration[7.0]
+  def change
+    create_table :admins do |t|
+      t.string :name
+      t.string :email
+      t.string :position
+      t.integer :private_number
+      t.string :password
```
Comment (YilKanda) [inline db/migrate/20220310235942_create_admins.rb]: Since if you implement a session management gem such as [Devise](https://github.com/heartcombo/devise) or [Clearance](https://github.com/thoughtbot/clearance), the gem will automatically generate a user table with the email field, which would generate inconsistency with the email data that you have already generated in your tables.

File: db/migrate/20220311005632_create_employees.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class CreateEmployees < ActiveRecord::Migration[7.0]
+  def change
+    create_table :employees do |t|
+      t.string :name
+      t.string :email
+      t.string :position
+      t.integer :private_number
+      t.references :branch, null: false, foreign_key: true
+
```
Comment (YilKanda) [inline db/migrate/20220311005632_create_employees.rb]: Since if you implement a session management gem such as [Devise](https://github.com/heartcombo/devise) or [Clearance](https://github.com/thoughtbot/clearance), the gem will automatically generate a user table with the email field, which would generate inconsistency with the email data that you have already generated in your tables.

File: db/migrate/20220310235942_create_admins.rb (hunk 2 above)
Comment (juperameza) [reply db/migrate/20220310235942_create_admins.rb]: Done on the new commit.

File: db/migrate/20220311005632_create_employees.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+      t.timestamps
+    end
+    add_index :employees, :email, unique: true
```
Comment (vyk2rr) [inline db/migrate/20220311005632_create_employees.rb:12]: I am going to suggest to think a little bit more about this approach:

So you added an index to `:employee` and `:admins` so it means:

```ruby
add_index :employees, :email, unique: true
add_index :admins, :email, unique: true
```

- you will have `:employees` using the `same name`, `but different email`:
  - name: `Perenganito Perez #1`, email: `user1111@example.com`
  - name: `Perenganito Perez #1`, email: `user2222@example.com` 
  - name: `Perenganito Perez #1`, email: `user3333@example.com` 
- and also within `:admins` you can also have the same user repeated:
  - name: `Perenganito Perez #1`, email: `user1111@example.com`
  
I think ^ that is going to bring difficulties in the future, so probably it is better to have a table to store roles, and have all users in just one table, and use roles instead to handle the different type of users (like admin, employee, etc) and only store the user id, in that way you won't repeat users data nor have difficulties handling this in the future. Unless that ^ is ok; You could also be more strict and have your index to also avoid having repeated names, but anyways you'll be ignoring that you have another table that have as well names and emails, so you need to check in both tables to avoid duplicates.

or use instead the `devise gem` https://github.com/heartcombo/devise to handle the user creation, permissions, and etc

Comment (juperameza) [reply db/migrate/20220311005632_create_employees.rb:12]: _you will have :employees using the same email, but different name:
name: Perenganito Perez #1, email: user1111@example.com
name: Perenganito Perez #1, email: user2222@example.com
name: Perenganito Perez #1, email: user3333@example.com
_
You said using the same email but different name but here you change the email and not the name. May you explain it a little bit more

**Review by VictorElizalde — APPROVED** [review]
LGTM
